packages/cli/src/rollup-plugins/typescript-declarations/common.ts

```diff
@@ -195,11 +195,32 @@ export const getDeclarationsForFile = async (
         (context) => (
           node
         ): import("typescript").Bundle | import("typescript").SourceFile => {
-          if (!visitModuleSpecifier) {
-            return node;
-          }
-
-          const cachedVisitModuleSpecifier = memoize(visitModuleSpecifier);
+          const replaceTsExtensions = (moduleSpecifier: string) => {
+            if (!moduleSpecifier.startsWith(".")) {
+              return moduleSpecifier;
+            }
+            return moduleSpecifier.replace(
+              /(\.d)?\.(ts|tsx|mts|cts)$/,
+              (match) => {
+                return {
+                  ".ts": ".js",
+                  ".tsx": ".js",
+                  ".mts": ".mjs",
+                  ".cts": ".cjs",
```

Comment:
I'm not quite sure how this would behave in older TS versions. It feels like a safe bet that those cases are not that popular today anyway and that we might triage the potential issues if somebody raises them. It also seems quite fair to say that using those extensions might require TS 4.8+ (or something like that) for package **consumers**.